Add feature flag to allow hiding the discussion tab for individual courses.

smarnach opened the pull request:
**Description**

Similar to the way the [Wiki tab can be hidden in the "Pages" section of a course in Studio](http://edx.readthedocs.org/projects/edx-partner-course-staff/en/latest/building_course/pages.html#hide-or-show-the-course-wiki-page), allow to hide the Discussion tab if this feature is enabled.  Hiding the tab will only make the link disappear from the tab bar, while manually navigating to the URL of the discussion forum will still work.

You can test this change on the [sandbox](http://sandbox2.opencraft.com:18010/): Navigate to Content > Pages on any course in Studio and click the eye icon on the Discussion tab, then go to the live version of the course and verify it has disappeared.  Note that you need to wait until the change was saved after clicking the icon.

This is a follow-up to #8086, which couldn't be re-opened.  As per discussion on [OSPR-599](https://openedx.atlassian.net/browse/OSPR-599), I put this behind a feature flag.  The implementation is completely different than the old one, since tabs have been refactored and moved into the individual Django apps.  The new implementation doesn't add a test, since it only adds a single assignment at class level, which can't be meaningfully unit-tested.
- - -
**Partner information** not an edX partner - 3rd party-hosted open edX instance
**JIRA Story** [OSPR-599](https://openedx.atlassian.net/browse/OSPR-599)
**Confluence / Product Asset** N/A
**Sandbox URL** [LMS](http://sandbox2.opencraft.com:/) [Studio](http://sandbox2.opencraft.com:18010/)
**Dependencies** N/A
**PR Author(s) Notes / To-Do** N/A
- - -
**Screenshots** 
The added icon is highlighted in this screenshot:
![image](https://cloud.githubusercontent.com/assets/249196/8106006/287e559c-103e-11e5-9cfb-d5e20c38e9ff.png)
- - -
**Reviewers**
Code: @mtyaka, (TBD); Product: (TBD)
- - -
**Settings**
```yaml
EDXAPP_FEATURES:
  ALLOW_HIDING_DISCUSSION_TAB: true
```

edx-webhook commented:
Thanks for the pull request, @smarnach! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request. 

To automatically create an OSPR issue for this pull request, just visit this link: http://openedx-webhooks.herokuapp.com/github/process_pr?number=8474&repo=edx%2Fedx-platform

sarina commented:
@smarnach OK, this makes sense to me, however I'm going to hand off to the TNL team to review because they know more about the tab refactoring than I do.

mtyaka commented:
Looks good :+1:

sarina commented:
@smarnach I asked the TNL team and this looks good, :+1:  Can I ask you to please rebase the PR? You can merge it when the build passes, or ask me to do it for you.

explorerleslie commented:
@sarina I'm unclear. It looks like this PR is ready for review, but the JIRA status is "waiting on author"

sarina commented:
@explorerleslie I took at look at this with Diana this morning and she gave a thumbs up. However it can't be merged because it has a conflict, so the author needs to rebase. Thus it's waiting on author, and then we can merge it.

explorerleslie commented:
@sarina sweet, thanks!

smarnach commented:
@sarina Thanks!  I rebased, added myself to the AUTHORS file (which got lost during my first patch), and will merge the PR myself once the build passes.
